src/antsibull/changelog.py

```diff
@@ -489,6 +503,258 @@ def __init__(
         self.collection_metadata = collection_metadata
 
 
+def _markup_to_rst(markup: str) -> str:
+    return _ansible_markup_to_rst(
+        _parse_ansible_markup(
+            markup,
+            _AnsibleMarkupContext(),
+            whitespace=_AnsibleMarkupWhitespace.KEEP_SINGLE_NEWLINES,
+        )
+    )
+
+
+def _get_removal_entry(  # noqa: C901, pylint:disable=too-many-branches
+    collection: str,
+    removal: RemovalInformation,
+    ansible_version: PypiVer,
+) -> tuple[ChangelogFragment, str] | None:
+    if (
+        removal.announce_version is None
+        or removal.announce_version.major != ansible_version.major
+    ):
+        return None
+
+    sentences = []
+    link = ""
+    if removal.discussion:
+        link = f" (`{removal.discussion} <{removal.discussion}>`__)"
+
+    if removal.reason == "deprecated":
+        sentences.append(f"The ``{collection}`` collection has been deprecated.")
+        sentences.append(
+            f"It will be removed from Ansible {removal.major_version} if no one"
+            f" starts maintaining it again before Ansible {removal.major_version}."
```

Comment:
Do we need the "if no one starts..." part if the collection was deprecated by upstream? If the collection maintainers themselves explicitly state they no longer intend to publish new releases, isn't that pretty final?

Reply:
At least in theory I guess it's possible that the (former) maintainers change their mind or give control to someone who stepped up to resurrect the collection, isn't it?

Not too probable, but possible. I could live with both removing this and keeping it as-is.

Reply:
> At least in theory I guess it's possible that the (former) maintainers change their mind or give control to someone who stepped up to resurrect the collection, isn't it?

Yes. Which actually happened recently with one of the network collections (https://forum.ansible.com/t/6244).

I guess it depends on the reason for deprecating it (if the service goes offline, there's no sense in un-deprecating it; but if it's simply "we don't have time for maintainenace and we don't have anyone to take it up", it can still happen).

Reply:
@gotmax23 I think this is a good example why we should keep the text as-is. What do you think?